File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/smart_pointer_overload_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'smart_pointer_overload'

include Smart_pointer_overload

f = Foo.new
b = Bar.new(f)


raise RuntimeError if f.test(3) != 1
raise RuntimeError if f.test(3.5) != 2
raise RuntimeError if f.test("hello") != 3

raise RuntimeError if b.test(3) != 1
raise RuntimeError if b.test(3.5) != 2
raise RuntimeError if b.test("hello") != 3

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/smart_pointer_overload_runme.py
from smart_pointer_overload import *

f = Foo()
b = Bar(f)


if f.test(3) != 1:
    raise RuntimeError
if f.test(3.5) != 2:
    raise RuntimeError
if f.test("hello") != 3:
    raise RuntimeError

if b.test(3) != 1:
    raise RuntimeError
if b.test(3.5) != 2:
    raise RuntimeError
if b.test("hello") != 3:
    raise RuntimeError
